cmd/api-mock/main.go

```diff
@@ -30,6 +36,52 @@ func Health(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+type ClustersEndpoint struct {
+	data    map[ClustersKey]string
+	dataMux sync.RWMutex
```

Comment:
Think the mutex has to be a pointer, as it can't be copied and you want to make sure you are referencing the same mutex accross all goroutines: https://stackoverflow.com/questions/67786469/use-mutex-or-mutex-pointer-in-golang

Reply:
right, tks!

Reply:
Isn't the solution to this to hold a pointer to a `ClustersEndpoint` struct instead?

```golang
type ClustersEndpoint struct {
  mux sync.Mutex
}
c := &ClustersEndpoint{}
c.mux.Lock() // <- this does not copy the mutex, right?
// Because .Lock is implemented on a pointer receiver: https://pkg.go.dev/sync#Mutex.Lock
```

Reply:
Josh is right.
The only reason to reference a Mutex via pointer is, if you are passing it around alone, which makes no sense.
Long post here: https://github.com/openshift/addon-operator/pull/125#discussion_r782007807

Reply:
Yes and all the data that the mutex is protecting, needs to be a pointer as well or the shallow copy will not work as expected. Which is the case for maps and slices, but if you have a bool or string in the mix, you will have to refactor to a pointer receiver anyway.